src/command/upload.ts

```diff
@@ -0,0 +1,142 @@
+import { Option, LeafCommand, Argument } from 'furious-commander'
+import { RootCommand } from './root-command'
+import * as FS from 'fs'
+import * as Path from 'path'
+import { sleep } from '../utils'
+import { Tag } from '@ethersphere/bee-js/dist/types'
+import { SingleBar, Presets } from 'cli-progress'
+import { bold, green, dim, red } from 'kleur'
+
+export class Upload extends RootCommand implements LeafCommand {
+  // CLI FIELDS
+
+  public readonly name = 'upload'
+
+  public readonly description = 'Upload file to Swarm'
+
+  @Argument({ key: 'path', describe: 'Path of the file (even directory)', demandOption: true })
+  public path!: string
+
+  // @Option({ key: 'encrypt', describe: 'Encrypt the passed data with a randomly generated key' })
+  // public encrypt!: boolean
+
+  @Option({ key: 'pin', type: 'boolean', describe: 'Preserve the uploaded data on the gateway node' })
+  public pin!: boolean
+
+  @Option({ key: 'recursive', alias: 'r', describe: 'Upload directory', default: true })
+  public recursive!: boolean
+
+  @Option({ key: 'tag-polling-time', describe: 'Waiting time in ms between tag pollings', default: 500 })
+  public tagPollingTime!: number
+
+  @Option({ key: 'tag-polling-trials', describe: 'After the given trials the tag polling will stop', default: 15 })
+  public tagPollingTrials!: number
+
+  @Option({
+    key: 'index-document',
+    describe: 'Default retrieval file on bzz request without provided filepath',
+    default: 'index.html',
+  })
+  public indexDocument!: string
+
+  @Option({
+    key: 'error-document',
+    describe: 'Default error file on bzz request without with wrong filepath',
+  })
+  public errorDocument!: string
+
+  // CLASS FIELDS
+
+  public hash!: string
+
+  public async run(): Promise<void> {
+    this.initCommand()
+    let tag = await this.bee.createTag()
+    let url: string
+
+    if (!FS.existsSync(this.path)) {
+      console.warn(bold().red(`Given filepath '${this.path}' doesn't exist`))
+
+      return
+    }
+
+    if (FS.lstatSync(this.path).isDirectory()) {
+      console.log('Starting to upload the given folder')
+      console.log(dim('Send data to the gateway Bee node...'))
+
+      if (this.pin) console.log(dim('Pin the uploaded data'))
+
+      this.hash = await this.bee.uploadFilesFromDirectory(this.path, this.recursive, {
+        indexDocument: this.indexDocument,
+        errorDocument: this.errorDocument,
+        tag: tag.uid,
+        pin: this.pin,
+      })
+      url = `${this.beeApiUrl}/bzz/${this.hash}`
+    } else {
+      console.log('Starting to upload the given file')
+      console.log(dim('Send data to the gateway Bee node...'))
+
+      this.hash = await this.bee.uploadFile(FS.createReadStream(this.path), Path.basename(this.path), {
+        tag: tag.uid,
+        pin: this.pin,
+      })
+      url = `${this.beeApiUrl}/files/${this.hash}`
+    }
+    console.log(dim('Data have been sent to the gateway node successfully!'))
```

Comment:
```suggestion
      console.log(dim('Send data to the Bee node...'))

      if (this.pin) console.log(dim('Pin the uploaded data'))

      this.hash = await this.bee.uploadFilesFromDirectory(this.path, this.recursive, {
        indexDocument: this.indexDocument,
        errorDocument: this.errorDocument,
        tag: tag.uid,
        pin: this.pin,
      })
      url = `${this.beeApiUrl}/bzz/${this.hash}`
    } else {
      console.log('Starting to upload the given file')
      console.log(dim('Send data to the Bee node...'))

      this.hash = await this.bee.uploadFile(FS.createReadStream(this.path), Path.basename(this.path), {
        tag: tag.uid,
        pin: this.pin,
      })
      url = `${this.beeApiUrl}/files/${this.hash}`
    }
    console.log(dim('Data have been sent to the Bee node successfully!'))
```

IMHO gateway implies public node that is hosted somewhere. I'd like to avoid this naming for consumer defined URLs